spock-specs/src/test/groovy/org/spockframework/util/DataVariableMapTest.groovy

```diff
@@ -0,0 +1,144 @@
+package org.spockframework.util
+
+import spock.lang.Specification
+
+import java.util.AbstractMap.SimpleEntry
+
+class DataVariableMapTest extends Specification {
+  def testee = new DataVariableMap(
+    ['a', 'b'],
+    1, 2
+  )
+
+  def 'modifying methods throw exception'() {
+    when:
+    method(testee)
+
+    then:
+    thrown(UnsupportedOperationException)
+
+    where:
+    _ | method
+    _ | { it.put('foo', _) }
+    _ | { it.remove(_) }
+    _ | { it.putAll(foo: _) }
+    _ | { it.clear() }
+    _ | { it.entrySet().add(new SimpleEntry<>('foo', _)) }
+    _ | { it.entrySet().remove(_) }
+    _ | { it.entrySet().addAll([new SimpleEntry<>('foo', _)]) }
+    _ | { it.entrySet().retainAll([]) }
+    _ | { it.entrySet().removeAll([]) }
+    _ | { it.entrySet().clear() }
+  }
+
+  def 'size should return proper result'() {
```

Comment:
Please use imperative style and not should
`size should return proper result` -> `size returns proper result`